### DISCUSSION TITLE
Clone a Project.

### DISCUSSION THREAD

**Discussion by Aging-Developer** [supabase/supabase#29879] [Questions]
Hi, I want to clone a project, from A to B, both hosted on Supabase.  I have read the guide for migrations: https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects

But it mentions having to install Supabase locally on order to do it, is this correct?  I need a local version even for cloning when both are hosted?  

I there an easier way?  i.e. copy schema, then data, then edge functions?

**Comment by silentworks** [comment]
This is the way. Other options will require more manual work, also it's always best to create projects using migrations and not using the dashboard directly, this way you can setup a new project easily. Data migration is normally a different thing all together, you will have to come up with a good strategy for yourself, the CLI allows you to dump the data from the database https://supabase.com/docs/reference/cli/supabase-db-dump click on the data only example on that page.

**Comment by stevengrimaldo** [comment]
+1 on the ability to clone a project

**Comment by Kevin-McIsaac** [comment]
As an average python programmer, with typical Github and  SQL experience, I'm really impressed with how easy it was to build complex LLM/Python/Streamlit app using  supabase as the vector store and RDBMS. It really lives up to the "build in a weekend" part of the mission statement.  

As part of my testing and production strategy ("deploy to millions), I intend to clone the dev environment (hosted) into new projects and  have hit the wall in terms of using the dashboard. 
1. The new "Restore to new project" looks perfect but requires physical backups and my database is less than 15gb so only has logical backups. I'd like to see this use  logical backups.
2. Branching would work well, especially for testing, but when I branch I get an empty project. This is may be because I didn't set up the database locally and run supabase init.  Can Branching be set up from from a hosted only DB? Is this documented?
3. The other methods require some combo of CLI, local DB and Docker, which is beyond my current skills or the time I'm willing to spend to make this work.

I'd like to see solutions to #1 and perhaps #2
